src/secret_storage.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the SecretStorage for managing the persistent secret storage for the Flask charm."""
+
+import secrets
+import typing
+
+from ops import Object, Relation, RelationDataContent, RelationEvent
+
+if typing.TYPE_CHECKING:
+    from charm import FlaskCharm
+
+
+class SecretStorage(Object):
+    """A class that manages secret keys required by the FlaskCharm.
+
+    Attrs:
+        is_initialized: True if the SecretStorage has been initialized.
+    """
+
+    # bandit think this is a password
+    _FLASK_SECRET_KEY_KEY = "flask_secret_key"  # nosec
+    _PEER_RELATION_NAME = "secret-storage"
+
+    def __init__(self, charm: "FlaskCharm"):
```

Comment:
This is a circular dependency, should the type hint be `CharmBase`?

Reply:
Updated, thanks!